### ISSUE TITLE
msn-pecan crash in msn_switchboard_free() at switchboard.c:262

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dequis/msn-pecan#158]
```
Msn-pecan has just crashed. See attached debug log + backtrace.

What version of the product are you using? On what operating system?
I'm using msn-pecan git version 0.0.18-52-gb1a4a7c, and Pidgin 2.6.0devel
rev cdbf018a5ef98397e81d541269550d9c6fded457 on 2009/06/04.

uname -a:
Linux cthulhu 2.6.28-12-generic #43-Ubuntu SMP Fri May 1 19:27:06 UTC 2009
i686 GNU/Linux

lsb_release -d
Description:    Ubuntu 9.04

```

Original issue reported on code.google.com by `fras...@gmail.com` on 5 Jun 2009 at 7:50

Attachments:
- [output3_cleaned.log](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-158/comment-0/output3_cleaned.log)

**Comment by GoogleCodeExporter** [comment]
```
I think I was receiving some messages sent to me when I was offline, when it 
crashed.
I'm not sure because my chat log didn't get saved yet, at the time of the crash.
```

Original comment by `fras...@gmail.com` on 5 Jun 2009 at 8:00

**Comment by GoogleCodeExporter** [comment]
```
Yes, you were about to receive a message. I have seen this one too... once.

I take it it doesn't happen very often.
```

Original comment by `felipe.contreras` on 6 Jun 2009 at 8:50
- Added labels: **Type-Defect**, **Priority-High**

**Comment by GoogleCodeExporter** [comment]
Original comment by `felipe.contreras` on 6 Jun 2009 at 1:35
- Added labels: **Stability**, **Rare**

**Comment by GoogleCodeExporter** [comment]
```
It crashed again, in msn_switchboard_free(). I think it is a different crash 
(reading
the debug log I don't see any offline messages received), but I attach the new 
debug log.
If it is actually a different crash, I will file a new one.
```

Original comment by `fras...@gmail.com` on 12 Jun 2009 at 8:59

Attachments:
- [output4_cleaned.log](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-158/comment-4/output4_cleaned.log)

**Comment by GoogleCodeExporter** [comment]
```
This crash seems to be frequent, in 30 minutes Adium+pecan (last commit) 
crashed two times due to this issue,
```

Original comment by `devid...@gmail.com` on 13 Jun 2009 at 12:36
- Removed labels: **Rare**

**Comment by GoogleCodeExporter** [comment]
```
@devidfil you need to post the whole backtrace, it might be a different issue 
that
crashes at the same point.
```

Original comment by `felipe.contreras` on 13 Jun 2009 at 12:42

**Comment by GoogleCodeExporter** [comment]
```
Ok, however it crashed again but this time the crash is in 
msn_switchboard_unref, debug log attached
```

Original comment by `devid...@gmail.com` on 13 Jun 2009 at 12:49

Attachments:
- [debugunref.txt](https://storage.googleapis.com/google-code-attachments/msn-pecan/issue-158/comment-7/debugunref.txt)

**Comment by GoogleCodeExporter** [comment]
```
@devidfil that's a different crash, it's triggered by a timeout, this one is
triggered by an incoming message.
```

Original comment by `felipe.contreras` on 13 Jun 2009 at 2:51

**Comment by GoogleCodeExporter** [comment]
Original comment by `felipe.contreras` on 20 Jun 2009 at 10:39
- Added labels: **Milestone-0.1.0**

**Comment by GoogleCodeExporter** [comment]
```
This is probably fixed in git master, please verify.

It will be released on 0.0.20.
```

Original comment by `felipe.contreras` on 6 Jul 2009 at 12:48
- Changed state: **Fixed**
